### PR TITLE
in filter tls < tlf < tlog; extension cleanup

### PR DESCRIPTION
**Pull request by toolslive**

### CODE REVIEW

**Comment by domsj** [comment]
streamline this priority with the code in `find_tlog_file` ?

File: src/tlog/compression_test.ml [hunk 1]
```diff
@@ -64,14 +64,14 @@ let test_compress_file which () =
...
-  let archive_name x = x ^ ".tls"
+let snappy =
+  let archive_name x = x ^ ".tlx"
   and compressor = Compression.Snappy
   and tlog_name a = Filename.chop_extension a
   in
   (archive_name, compressor, tlog_name)
 
-let snappy =
+let bz2 =
```
Comment (domsj) [inline src/tlog/compression_test.ml:74]: hehe

File: src/tlog/compression.ml [hunk 2]
```diff
@@ -44,117 +44,138 @@ let uncompress_bz2 b =
 let compress_snappy b = Snappy.compress b
 let uncompress_snappy b = Snappy.uncompress b
 
+
+let format_snappy = "snappy"
+let read_format ic = function
```
Comment (domsj) [inline src/tlog/compression.ml:49]: So this code is used to verify that the right decompressor is being used when reading a file, right?
But shouldn't the format be read to be able to select the right decompressor in the first place?

**Comment by domsj** [comment]
Except for the comment about `read_format` this looks good to me.

File: src/tlog/compression.ml (hunk 2 above)
Comment (toolslive) [inline src/tlog/compression.ml:49]: putting bz2 inside .tlx too will be left for later.

**Pull request merged**